client/client.go

```diff
@@ -709,12 +708,90 @@ func (ec *SDKClient) GetGasPrice(
 			newGasPrice.Int(gasPrice)
 		}
 	} else {
-		log.Println("Setting existing gas price")
 		gasPrice = input.GasPrice
 	}
 	return gasPrice, nil
 }
 
+func (ec *SDKClient) GetGasTipCap(ctx context.Context, input Options) (*big.Int, error) {
+	if input.GasTipCap == nil {
+		var hex hexutil.Big
+		if err := ec.CallContext(ctx, &hex, "eth_maxPriorityFeePerGas"); err != nil {
+			return nil, err
+		}
+
+		return hex.ToInt(), nil
+	}
+
+	return input.GasTipCap, nil
+}
+
+func (ec *SDKClient) GetGasFeeCap(ctx context.Context, input Options, gasTipCap *big.Int) (*big.Int, error) {
+	if input.GasFeeCap == nil {
+		baseFee, err := ec.GetBaseFee(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		if baseFee != nil {
+			// Calculate max fee per gas
+			// Formula: GasFeeCap = max(BaseFeeMultiplier * BaseFee, BaseFeeFloor) + GasTipCap
+			baseFeeFloor := getBaseFeeFloor(ec.rosettaConfig)
+			baseFeeMultiplier := getBaseFeeMultiplier(ec.rosettaConfig)
+			adjustedBaseFee := new(big.Int).Mul(baseFee, baseFeeMultiplier)
+			gasFeeCap := new(big.Int).Set(bigIntMax(adjustedBaseFee, baseFeeFloor))
+			gasFeeCap.Add(gasFeeCap, gasTipCap)
+
+			fmt.Println(baseFeeFloor)
+			fmt.Println(baseFeeMultiplier)
+
+			return gasFeeCap, nil
+		}
+	}
+
+	return input.GasFeeCap, nil
+}
+
+func getBaseFeeFloor(rosettaConfig configuration.RosettaConfig) *big.Int {
+	baseFeeFloor := big.NewInt(100)
+	if rosettaConfig.BaseFeeFloor != nil {
+		baseFeeFloor = rosettaConfig.BaseFeeFloor
+	}
+
+	return baseFeeFloor
+}
+
+func getBaseFeeMultiplier(rosettaConfig configuration.RosettaConfig) *big.Int {
+	baseFeeMultiplier := big.NewInt(2)
```

Comment:
actually we need the hardcoded default value in rosetta layer as those values are in our own formula (i.e. GasFeeCap = max(BaseFeeMultiplier * BaseFee, BaseFeeFloor) + GasTipCap), given we also need to support configuring them in env var so if we don't put hardcoded default value the code path of configuring them will be broken if downstream service doesn't configure them (those env var should be optional for downstream services)

Reply:
Let's move the magic number to a file in sdk. No magic number in an impl file.

Reply:
sure, will move default values from our client util func to configuration const, also fwiw, those are only for default values, configuring those values should be still from downstream services

Reply:
well, i think it makes more sense to remove the magic numbers and in rosetta layer the formula becomes the generic EIP-1559, which is GasFeeCap = BaseFee + GasTipCap, so that the rosetta is EIP-1559 neutral and chain can modify it from chain specific layer according to chain property (i.e. EIP-1559 elasticity, etc)